functional/FunctionalTest.java

```diff
@@ -247,6 +247,25 @@ public static void putObject_test6() throws Exception {
   }
 
   /**
+   * Test: putObject(String bucketName, String objectName, String contentType, InputStream body).
+   */
+  public static void putObject_test7() throws Exception {
+    System.out.println("Test: putObject(String bucketName, String objectName, InputStream body, "
+                       + "String contentType)");
+    String filename = createFile(3 * MB);
+    InputStream is = Files.newInputStream(Paths.get(filename));
+    client.putObject(bucketName, filename, is, customContentType);
+    is.close();
+    Files.delete(Paths.get(filename));
+    ObjectStat objectStat = client.statObject(bucketName, filename);
+    if (!customContentType.equals(objectStat.contentType())) {
+      throw new Exception("[FAILED] Test: putObject(String bucketName, String objectName, String contentType, "
+                          + "long size, InputStream body)");
+    }
+    client.removeObject(bucketName, filename);
+  }
+
```

Comment:
Done.
As mentioned in the commit log, for unknown sized stream, multipart size is not 5MiB, it is 525MiB.